### ISSUE TITLE
Reorganize CGNS repository structure

### ISSUE THREAD

**Issue by MicK7** [CGNS/CGNS#798]
We should adopt a similar structure to hdf5 repos .
This would reorganize the repos like this:

- src
  - adf
  - adfh
  ... *.c *.h
- test
- testpar
- doxygen
- tools
- utils
- config
- UserExamples

This would mean that current tests that are under src would be moved up.
The cgnstools, utils, and tools would also move up.

**Comment by MicK7** [comment]
@brtnfld what's your opinion ?

**Comment by brtnfld** [comment]
I'm fine with the reorganization. Do you plan on moving the Autotools files to the top level, too? I always thought it strange that these files were not located there. I would call it just examples.

**Comment by MicK7** [comment]
> I'm fine with the reorganization. Do you plan on moving the Autotools files to the top level, too? I always thought it strange that these files were not located there. I would call it just examples.

Yes Autotools should be moved as well. I was thinking of waiting fot the 4.5.0 release before starting any change.

**Comment by brtnfld** [comment]
We should also introduce a clang-format spec so that all the C code conforms to a specific formatting standard. It is probably easiest to do this when the number of active branches is minimal.
